### PR TITLE
[FEAT] Spring Blossom Banner

### PR DESCRIPTION
**Pull request by Rofrtd**
# Description

This PR implements the new season banner sale for Block Bucks

Fixes #issue

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Added unit tests

# Checklist:

- [ ] Title of the PR is relevant and is prefixed with [FEAT], [CHORE] or [FIX]
- [ ] Screenshot if it includes any UI changes
- [ ] I have read the contributing guidelines and agree to the T&Cs
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/features/game/types/images.ts [hunk 1]
```diff
@@ -2386,6 +2391,10 @@ export const ITEM_DETAILS: Items = {
     description:
       "The Kraken is here! The mark of a participant in the Catch the Kraken Season.",
   },
+  "Spring Blossom Banner": {
+    image: catchTheKrakenBanner,
```
Comment (brian-man) [inline src/features/game/types/images.ts:2395]: Image needs updating.

File: src/features/game/types/seasons.ts [hunk 2]
```diff
@@ -31,19 +36,22 @@ export type SeasonalTicket =
...
   | "Crow Feather"
-  | "Mermaid Scale";
+  | "Mermaid Scale"
+  | "Spring Blossom Ticket";
 
-type SeasonalBanner =
-  | "Solar Flare Banner"
+export type SeasonalBanner =
   | "Dawn Breaker Banner"
+  | "Solar Flare Banner"
```
Comment (brian-man) [inline src/features/game/types/seasons.ts:44]: Why is `Solar Flare Banner` being moved?

File: src/features/game/types/images.ts (hunk 1 above)
Comment (Rofrtd) [reply src/features/game/types/images.ts:2395]: Yes, waiting for assets.

File: src/features/game/events/landExpansion/bannerPurchased.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  const { bumpkin, inventory } = stateCopy;
+
+  if (bumpkin === undefined) {
+    throw new Error("You do not have a Bumpkin");
+  }
+
+  if (inventory[action.name] !== undefined) {
+    throw new Error("You already have this banner");
+  }
+
```
Comment (adamhannigan) [inline src/features/game/events/landExpansion/bannerPurchased.ts:37]: Consider also asserting `action.name` - we want to ensure it is a valid banner and not some other item in the game.

E.g. `if (!(name in BANNERS))`

**Review by adamhannigan — COMMENTED** [review]
Nice work mate!

Just facing a small issue when I attempt to buy the banner with Block Bucks. It is trying to make me connect a wallet first - we will want to ensure this works for Google players who don't have a wallet 👍 

Also, I would suggest replacing the button text - "Buy Now 50" -> "Buy Now"

Screenshot 2024-01-11 at 10 31 23 am

**Comment by adamhannigan** [comment]
Screenshot 2024-01-11 at 10 35 10 am

Could we also make this more of a confirmation screen. Now that we are doing it offchain, it is much easier and quicker for a player to purchase the banner (which means they may accidentally purchase it).

"Are you sure you want to purchase the banner?"

"50 Block Bucks"

"Cancel" - "Confirm"

**Review by adamhannigan — APPROVED** [review]
LGTM

**Pull request merged**